src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
I've asked this before, but is the orphaning really necessary?  Once 80% support Segwit2x, isn't it likely 15 of the remaining 20% support BIP141?

Forgive me if I'm misreading and the code already does this, but ideally I'd think orphaning should start (if at all) when bit 4 is **activated,** but bit 1 signaling (w/o orphaning) as soon as bit 4 is **locked in.**  I see no reason at all why the latter should wait till bit 4 is activated.